modules/RegeneronGeneBurden.py

```diff
@@ -0,0 +1,326 @@
+        # Given the name and the associations it reports, it makes sense that it is MARCHF8 (ENSG00000165406).
+        # For the moment we will remove them until this is fixed.
+        .filter(~col("Gene").contains("2021"))
+        .distinct()
+        .persist()
+    )
+
+    evd_df = parse_regeneron_evidence(regeneron_df)
+    if evd_df.filter(col('resourceScore') == 0).count() != 0:
+        logging.error("There are evidence with a P value of 0.")
+        raise AssertionError(
+            f"There are {evd_df.filter(col('resourceScore') == 0).count()} evidence with a P value of 0."
+        )
+
+    if not 7000 < evd_df.count() < 8000:
+        logging.error(f"REGENERON number of evidence are different from expected: {evd_df.count()}")
+        raise AssertionError("AZ PheWAS Portal number of evidence are different from expected.")
+
+    if evd_df.filter(col('studySampleSize').isNull()).count() != 0:
+        logging.error(
+            f"There are {evd_df.filter(col('studySampleSize').isNull()).count()} evidence with a null sample size."
+        )
+        raise AssertionError(f"Study sample size is missing for some associations.")
+    logging.info(f"{evd_df.count()} evidence strings have been processed.")
+
+    return evd_df
+
+
+def parse_regeneron_evidence(regeneron_df: DataFrame) -> DataFrame:
+    """
+    Parse REGENERON's ExWAS analyses.
+    Args:
+        regeneron_df: DataFrame with the Regeneron data
+    Returns:
+        evd_df: DataFrame with the Regeneron data following the t/d evidence schema.
+    """
+    to_keep = [
+        "datasourceId",
+        "datatypeId",
+        "targetFromSourceId",
+        "diseaseFromSource",
+        "diseaseFromSourceMappedId",
+        "pValueMantissa",
+        "pValueExponent",
+        "beta",
+        "betaConfidenceIntervalLower",
+        "betaConfidenceIntervalUpper",
+        "oddsRatio",
+        "oddsRatioConfidenceIntervalLower",
+        "oddsRatioConfidenceIntervalUpper",
+        "resourceScore",
+        "ancestry",
+        "ancestryId",
+        "literature",
+        "projectId",
+        "cohortId",
+        "studyId",
+        "studySampleSize",
+        "studyCases",
+        "studyCasesWithQualifyingVariants",
+        "statisticalMethod",
+        "statisticalMethodOverview",
+    ]
+
+    get_exponent_udf = udf(get_exponent, IntegerType())
+    get_mantissa_udf = udf(get_mantissa, DoubleType())
+
+    return (
+        regeneron_df.withColumn("datasourceId", lit("gene_burden"))
+        .withColumn("datatypeId", lit("genetic_association"))
+        .withColumn("literature", array(lit("34662886")))
+        .withColumn("projectId", lit("REGENERON"))
+        .withColumn("cohortId", lit("UK Biobank 450k"))
+        .withColumnRenamed("Gene", "targetFromSourceId")
+        .withColumnRenamed("Trait", "diseaseFromSource")
+        .withColumnRenamed("MAPPED_TRAIT", "diseaseFromSourceMappedId")
```

Comment:
If we'll update the script to handle multiple traits for a study, this column will be a list and needs to be exploded at this point.

Reply:
Fixed.